torch/csrc/utils/tensor_new.cpp

```diff
@@ -21,72 +17,89 @@
 #include "torch/csrc/utils/tensor_conversion_dispatch.h"
 #include "torch/csrc/utils/tensor_numpy.h"
 
-static const int MAX_DIMS = 128;
+#include <ATen/ATen.h>
+#include <ATen/Error.h>
+#include <ATen/optional.h>
 
-using namespace at;
+#include <stdexcept>
+#include <vector>
+
+// Certain types have the same name in torch:: and at::, so better use
```

Comment:
this part is actually quite fragile. Why not name classes separately? E.g. it could be VariableTensorOptions instead of torch::TensorOptions

Reply:
I might want to call it `VariableOptions`, yes